kyuubi-server/src/main/scala/org/apache/kyuubi/events/KyuubiSessionEvent.scala

```diff
@@ -55,7 +55,7 @@ case class KyuubiSessionEvent(
 object KyuubiSessionEvent {
   def apply(session: KyuubiSessionImpl): KyuubiSessionEvent = {
     assert(KyuubiServer.kyuubiServer != null)
-    val serverIP = KyuubiServer.kyuubiServer.connectionUrl
+    val serverIP = KyuubiServer.kyuubiServer.frontendServices.head.connectionUrl
```

Comment:
this is quite hacky, but not related to this change

Reply:
I think we can add a method in `Utils` which check `frontendServices` only has one element and extract `connectionUrl` from the it.

Reply:
that's hacky too